### PR TITLE
Add Mark Unmark Commands

### LINKED ISSUE #16: Add mark/unmark command
As a tutor, I can mark students who are present and absent from class so that I can keep track of attendance using this application.

### PR DESCRIPTION
**Pull request by comicalromance**
- Create two new commands (Mark and Unmark) which allow the user to mark attendance for students
- Create new attribute Attendance to Person, and updated subsequent Storage + Serialization + UI
- Add corresponding JUnit tests

# Notes
- To serialize Attendance, className and hasAttended (absent / present) were concatenated into a string and stored as json. Could be cleaner to use nested json instead to represent Attendance
Close #16

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddTagCommand.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public class AddTagCommand extends Command {
     /**
      * Constructs AddTagCommand that will add specified tag to all persons in the
      * displayed list.
-     * @param index of the person in the filtered person list to add the tag
```
Comment (Piyotato) [inline src/main/java/seedu/address/logic/commands/AddTagCommand.java:68]: Should this fix have been done on another branch instead?

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -11,7 +11,7 @@
 import seedu.address.model.person.Person;
 
 /**
- * Deletes a person identified using it's displayed index from the address book.
```
Comment (Piyotato) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:14]: Should this fix have been done on another branch instead?

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Person personToEdit = lastShownList.get(index.getZeroBased());
+        PersonData personData = new PersonData();
+        personData.setName(personToEdit.getName());
+        personData.setPhone(personToEdit.getPhone());
+        personData.setEmail(personToEdit.getEmail());
+        personData.setAddress(personToEdit.getAddress());
+        personData.setTags(personToEdit.getTags());
```
Comment (Piyotato) [inline src/main/java/seedu/address/logic/commands/UnmarkCommand.java:64]: I just realized that there's a lot of duplicated code when reconstructing PersonData from Person; should we make a method in Person that dumps its PersonData?

**Review by Piyotato — APPROVED** [review]
LGTM, how about the others?

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java (hunk 3 above)
Comment (highorbit25) [reply src/main/java/seedu/address/logic/commands/UnmarkCommand.java:64]: Perhaps we can do that for v1.2b?

File: src/main/java/seedu/address/logic/commands/AddTagCommand.java (hunk 1 above)
Comment (comicalromance) [reply src/main/java/seedu/address/logic/commands/AddTagCommand.java:68]: True but I just wanted to fix a bunch of more trivial documentation issues

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java (hunk 2 above)
Comment (comicalromance) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:14]: Same for this

File: src/main/java/seedu/address/logic/commands/UnmarkCommand.java (hunk 3 above)
Comment (comicalromance) [reply src/main/java/seedu/address/logic/commands/UnmarkCommand.java:64]: Makes sense, can do that in a separate branch!

**Pull request merged**